website/docs/r/resource_group_template_deployment.html.markdown

```diff
@@ -15,31 +15,47 @@ Manages a Resource Group Template Deployment.
 ## Example Usage
 
 ```hcl
+locals {
+  vnet_name = "example-vnet"
+}
+
 resource "azurerm_resource_group_template_deployment" "example" {
   name                = "example-deploy"
   resource_group_name = "example-group"
   deployment_mode     = "Complete"
-  template_content    = <<TEMPLATE
+  parameters_content = jsonencode({
+    "vnetName" = {
+      value = local.vnet_name
+    }
+  })
+  template_content = <<TEMPLATE
 {
-  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-  "contentVersion": "1.0.0.0",
-  "parameters": {},
-  "variables": {},
-  "resources": [
-    {
-      "type": "Microsoft.Network/virtualNetworks",
-      "apiVersion": "2020-05-01",
-      "name": "acctest-network",
-      "location": "[resourceGroup().location]",
-      "properties": {
-        "addressSpace": {
-          "addressPrefixes": [
-            "10.0.0.0/16"
-          ]
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vnetName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the VNET"
+            }
         }
-      }
-    }
-  ]
+    },
+    "variables": {},
+    "resources": [
+        {
+            "type": "Microsoft.Network/virtualNetworks",
+            "apiVersion": "2020-05-01",
+            "name": "acctest-network",
```

Comment:
for the purposes of this example we should probably use this too, e.g.

```suggestion
            "name": "[variables("vnetName")]",
```

Reply:
Agree, good call.